src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -539,12 +540,102 @@ void Subgraph::createPrimitive() {
         initPluginBlockedShapes();
         initAttributes();
         initStartOffsets();
+        initBroadcastableInputs();
         optimizeIR();
     }
 
     Node::createPrimitive();
 }
 
+void Subgraph::initBroadcastableInputs() {
+    // Snippets supports tokenization of the following operations:
+    // - Unary, Binary and Ternary (Select) Elementwise ops
+    // - Softmax, MatMul, Transpose, GroupNorm
+    // Binary Elementwise ops (+ Select) requires explicit Broadcast op
+    // on inputs if broadcasting of latest dimensions is needed.
+    // These ops will be start points of DFS - need to go to Parameters and update `broadcastable_inputs_map`.
+    // We iterates through all ops by execution order. So if we already analyzied some op in the input branch - skip this branch.
+    // However, there some ops which can change `processing_dim_idx`:
+    // - Transpose has order which changes `processing_dim_idx`. But Transpose can be only before Parameters
+    // - MatMul's first input doesn't affect output latest dimension - skip this branch.
+    //   Also MatMul has `transposed_b` which changes `processing_dim_idx`
+    broadcastable_inputs.clear();
+    const auto& body = subgraph_attrs->snippet->body_ptr();
+    if (body->get_parameters().size() < 2)
+        return;
+
+    const auto& ops = body->get_ordered_ops();
+    std::set<std::shared_ptr<ov::Node>> visited_ops = {};
+    for (const auto& op : ops) {
+        if (!ov::snippets::lowered::pass::InsertBroadcastMove::is_broadcasting_supported(op))
+            continue;
+
+        size_t processing_dim_idx = 0;
+
+        // DFS
+        std::stack<std::shared_ptr<ov::Node>> nodes_to_calculate;
+        nodes_to_calculate.push(op);
+        while (!nodes_to_calculate.empty()) {
+            auto current_node = nodes_to_calculate.top();
+            nodes_to_calculate.pop();
+
+            if (const auto& param = ov::as_type_ptr<ov::op::v0::Parameter>(current_node)) {
+                const auto consumers = param->get_output_target_inputs(0);
+                if (std::any_of(consumers.cbegin(), consumers.cend(),
+                                [](const ov::Input<ov::Node>& in) { return ov::is_type<ov::op::v1::Transpose>(in.get_node()); })) {
+                    OPENVINO_ASSERT(consumers.size() == 1, "Incorrect count of outputs of Parameter!");
+                    const auto transpose = consumers.begin()->get_node();
+                    std::vector<size_t> order;
+                    const auto& constant = ov::as_type_ptr<const opset1::Constant>(transpose->get_input_node_shared_ptr(1));
+                    OPENVINO_ASSERT(constant, "Unsupported order node of Transpose");
+                    order = constant->cast_vector<size_t>();
+                    if (order.empty()) {
+                        order.resize(transpose->get_output_partial_shape(0).size());
+                        std::iota(order.rbegin(), order.rend(), 0);
+                    }
```

Comment:
Can the transpose order be really empty?

Reply:
Yep! [The spec](https://docs.openvino.ai/2023.3/openvino_docs_ops_movement_Transpose_1.html).